### ISSUE TITLE
Bugsnag plugin broken with android plugin 1.5.+

### ISSUE THREAD

**Issue by redwarp** [bugsnag/bugsnag-android-gradle-plugin#8]
I noticed that on crashes, the traces would still be obfuscated for a prod release with proguard.
So I checked my apk with aapt, and in the final apk, the meta-data is not set.

So I dug a bit deeper, read the plugin file, and saw that, you do that in https://github.com/bugsnag/bugsnag-android-gradle-plugin/blob/master/src/main/groovy/com/bugsnag/android/gradle/BugsnagPlugin.groovy

```
if (variant.getObfuscation() == null) {
    return
}
```

Problem: I use gradle android plugin 1.5.+, and plan to jump on the version 2.0.0 whenever it's ready (to benefit from all the cool features, such as code hot swapping and everything), and `variant.getObfuscation()` does not work anymore...

It looks like hockeyapp had the same issue, and here is how they solve it: https://github.com/x2on/gradle-hockeyapp-plugin/issues/74

If you can fix it as well it would be really useful, and would avoid me the hassle of having to create a custom gradle task that do a curl request to upload the mapping...

Cheers.

**Comment by loopj** [comment]
Hey @redwarp thanks so much for finding this and pointing us towards a fix!

I've just [released v1.2.2](https://github.com/bugsnag/bugsnag-android-gradle-plugin/releases/tag/v1.2.2) which should be making its way into maven central now, please let me know if this fixes your issue!

**Comment by redwarp** [comment]
Hey, thanks for your quick answer, really appreciated. Sadly, this somehow seems to make maters worse, as with version 1.2.2, I can't sync the project with Android Studio anymore.

I get the following error:
```
Error:Task with name 'proguardProdRelease' not found in project ':YelpBiz'.
```

To reproduce:
- Clean the build directory (like, remove it)
- Synchronize Android studio, clicking `Sync Project with Gradle Files` <- that works
- With the Build menu, click `Generate signed APK...` and, well, generate a signed APK.
- If you check the APK, the BUILD_UIID is not in the manifest
- Synchronize Android studio, clicking `Sync Project with Gradle Files`
- That should fail!

It look like this line, at the end of `BugsnagPlugin.groovy`, is creating the issue:
```
bugsnagTask.dependsOn project.tasks["proguard${variantName}"]
```
Here are my conclusions:

- The method `variant.getMappingFile()?.exists()` actually checks if the map file is present. So if the build was not cleaned, and we have a prior build with proguard, then this will return true.
- When proguard is not enabled, then `project.tasks["proguard${variantName}"]` does not exist, and so `bugsnagTask.dependsOn` fails, and it's not possible to do the sync, as the gradle build chain is not consistent.
- If not maps is present before, then the bugsnag plugin does not create the tasks, and the build will not have the meta data in the manifest, and will not upload the map file to your server.

So the condition `is there a mapping file` does not really work with this scenario. Maybe `project.tasks["proguard${variantName}"] != null` could serve as a condition instead? (Not sure, untested)

**Comment by d4rken** [comment]
What's the status on this?

**Comment by d4rken** [comment]
@loopj Great, let me know if you need any further information on the issue, but so far I can confirm what @redwarp described.

Now I'm pondering whether to wait for the fix or use the manual "curl" based upload. :smile:

**Comment by loopj** [comment]
This should be fixed in #10 - hopefully we can release this tomorrow

**Comment by loopj** [comment]
Hey all, this fix has been released in 2.0.0 of the plugin, should be hitting maven central soon, thanks for all the help!

**Comment by redwarp** [comment]
Hum, I think it never reached maven central...

**Comment by d4rken** [comment]
@redwarp I agree.
@loopj http://mvnrepository.com/artifact/com.bugsnag/bugsnag-android-gradle-plugin

**Comment by loopj** [comment]
@redwarp @d4rken looks like there was a pom.xml validation error which stopped this from propagating to maven central, just fixed and re-pushed:

https://search.maven.org/#artifactdetails%7Ccom.bugsnag%7Cbugsnag-android-gradle-plugin%7C2.0.0%7Cjar
https://github.com/bugsnag/bugsnag-android-gradle-plugin/releases/tag/v2.0.0

Apologies for the delay!

**Comment by loopj** [comment]
Just FYI - 2.0.0 was built targeting Java 1.8 by mistake, just released 2.0.1 which fixes this.
